File: transceiver_interface.m
function [Const_Tx_real, Const_Tx_imag, Const_Rx_real, Const_Rx_imag] = transceiver_interface(Freq_THz,Comm_dist_init,Skin_thickness,Tissue_thickness,Vessel_thickness,blood_speed,mod_order_int,bits,A)
    Const_Tx_real = zeros(1,length(bits));
    Const_Tx_imag = zeros(1,length(bits));
    Const_Rx_real = zeros(1,length(bits));
    Const_Rx_imag = zeros(1,length(bits));

    coder.updateBuildInfo('addSourceFiles','terahertz.c');
    coder.cinclude('terahertz.h'); 
    coder.ceval('transceiver',Freq_THz,Comm_dist_init,Skin_thickness,Tissue_thickness,Vessel_thickness,blood_speed,mod_order_int,coder.rref(bits),length(bits),A,coder.rref(Const_Tx_real),coder.rref(Const_Tx_imag),coder.wref(Const_Rx_real),coder.wref(Const_Rx_imag));
end

File: codegen/mex/transceiver_interface/transceiver_interface.c
/*
 * Academic License - for use in teaching, academic research, and meeting
 * course requirements at degree granting institutions only.  Not for
 * government, commercial, or other organizational use.
 *
 * transceiver_interface.c
 *
 * Code generation for function 'transceiver_interface'
 *
 */

/* Include files */
#include "transceiver_interface.h"
#include "rt_nonfinite.h"
#include "terahertz.h"
#include <string.h>

/* Function Definitions */
void transceiver_interface(
    const emlrtStack *sp, real_T Freq_THz, real_T dist_init_x,
    real_T Skin_thickness, real_T Tissue_thickness, real_T Vessel_thickness,
    real_T blood_speed, real_T mod_order_int, const real_T bits[64000],
    real_T bit_rate, real_T A, real_T Const_Tx_real[64000],
    real_T Const_Tx_imag[64000], real_T Const_Rx_real[64000],
    real_T Const_Rx_imag[64000], real_T nanosensor_pos_x[64000])
{
  (void)sp;
  memset(&Const_Tx_real[0], 0, 64000U * sizeof(real_T));
  memset(&Const_Tx_imag[0], 0, 64000U * sizeof(real_T));
  transceiver(Freq_THz, dist_init_x, Skin_thickness, Tissue_thickness,
              Vessel_thickness, blood_speed, mod_order_int, &bits[0], 64000.0,
              bit_rate, A, &Const_Tx_real[0], &Const_Tx_imag[0],
              &Const_Rx_real[0], &Const_Rx_imag[0], &nanosensor_pos_x[0]);
}

/* End of code generation (transceiver_interface.c) */
